spirv_msl.cpp

```diff
@@ -2958,6 +3163,23 @@ string CompilerMSL::to_function_args(uint32_t img, const SPIRType &imgtype, bool
 		farg_str += to_expression(sample);
 	}
 
+	if (msl_options.swizzle_texture_samples && imgtype.image.sampled == 1 && imgtype.image.dim != DimBuffer &&
+	    (!is_gather || !imgtype.image.depth))
+	{
+		// Add the swizzle constant from the swizzle buffer.
+		if (!is_gather)
+			farg_str += ")";
+		// Get the original input variable for this image.
+		uint32_t img_var = img;
+		if (meta[img].image)
+			img_var = meta[img].image;
+		if (auto *var = maybe_get_backing_variable(img_var))
+			img_var = var->self;
+		auto &aux_type = expression_type(aux_buffer_id);
+		farg_str += ", " + to_name(aux_buffer_id) + "." + to_member_name(aux_type, k_aux_mbr_idx_swizzle_const) + "[" +
```

Comment:
This will still only work for images which are passed implicitly as a global, not as a parameter. But that'd be too insane to deal with for now. Could you add a check here, and error out if it cannot find a valid resource index? (i.e. it's a parameter, not a global alias).